### PR TITLE
fix: robots and sitemap for prod build

### LINKED ISSUE #48: [Bug] Environment Variables
https://github.com/Jam3/nyg-nextjs/pull/45

### LINKED ISSUE #81: [Bug] Update nextjs-sitemap-generator to 1.3.0
## Environment

Run the below command and paste the result here:

```
npx envinfo --system --npmPackages react* --binaries --npmGlobalPackages react* --browsers
```

### LINKED ISSUE #86: [Bug] fix robots generator
## Describe the bug

The commend `npm run build:prod:static`, it does not generate `robots.txt` and `sitemap.xml`.

## To Reproduce
Run the commend `npm run build:prod:static` locally, and check `out` folder. 

## Expected behaviour

1. Running `npm run build:develop:static` or `npm run build:staging:static` should use the default robots.txt file that is stored in `public` folder
2. Running `npm run build:prod:static` must generate robots.txt for PROD use only from the `robots-txt-generator` script.

## Environment

Run the below command and paste the result here:

```
npx envinfo --system --npmPackages react* --binaries --npmGlobalPackages react* --browsers
```

### PR DESCRIPTION
**Pull request by DonghyukJacobJang**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Code style update
- [ ] Refactor (refactoring or adding test which isn't a fix or add a feature)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Did you test your solution?**

- [ ] I lightly tested it in one browser
- [ ] I deeply tested it in several browsers
- [ ] I wrote tests around it (unit tests, integration tests, E2E tests)

## Problem Description
#81 and #86 

## Side Effects, Risks, Impact

- [ ] N/A

**Aditional comments:**

This should fix #48.
This should fix #81.
This should fix #86.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -108,7 +108,7 @@
     "next-compose-plugins": "2.2.0",
     "next-images": "^1.4.1",
     "next-optimized-images": "2.6.2",
-    "nextjs-sitemap-generator": "^1.0.0",
```
Comment (DonghyukJacobJang) [inline package.json:111]: this old version has a bug when dealing with trailing slash in the sitemap.

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "start": "next start",
     "\n ######################  NPM Secondary Pipeline  ######################": "",
     "build:ssr": "next build",
-    "build:prod:static": "next build && next export && cross-env CI_ENV=production npm-run-all robots-txt-generator sitemap-generator",
```
Comment (DonghyukJacobJang) [inline package.json:18]: this has arisen from multiple projects that our `develop` and `staging` envs using `.env.production` when building and deploying. so we are renaming `.env.production` to `.env.prod`

File: scripts/nextjs-sitemap-generator.js [hunk 3]
```diff
@@ -15,11 +15,11 @@ if (process.env.NEXT_PUBLIC_WEBSITE_SITE_URL) {
     // fr: 'https://example.com/fr'
     // },
     baseUrl: process.env.NEXT_PUBLIC_WEBSITE_SITE_URL.replace('://', '://www.'),
-    ignoredPaths: ['admin'],
+    ignoredPaths: ['/404', '/unsupported', '/api'],
     pagesDirectory: path.resolve(__dirname + '/../src/pages'),
     targetDirectory: 'out/',
     // NOTE: enabled nextConfigPath option if exportPathMap is enabled in next.config.js
-    // nextConfigPath: path.resolve(__dirname + '/../next.config.js'),
+    nextConfigPath: path.resolve(__dirname + '/../next.config.js'),
```
Comment (DonghyukJacobJang) [inline scripts/nextjs-sitemap-generator.js:22]: necessary for adding a trailing slash to the url

**Review by neo — APPROVED** [review]
🚀

**Comment by neo** [comment]
This should fix #48, #81 and #86

File: scripts/robots-txt-generator.js [hunk 4]
```diff
@@ -7,15 +7,15 @@ require('dotenv').config({
...
+if (process.env.CI_ENV === 'prod' && process.env.NEXT_PUBLIC_WEBSITE_SITE_URL) {
   console.log(chalk.cyan('\n###################### robots.txt ######################\n'));
 
   robotstxt({
     policy: [
       {
         userAgent: '*',
         allow: '/',
-        disallow: ['/_next']
+        disallow: ['/_next', '/aseets']
```
[the remark this answers is not in the document]
Comment (DonghyukJacobJang) [reply scripts/robots-txt-generator.js]: typo!

**Review by neo — APPROVED** [review]
reapproved!

**Pull request merged**